Agents/AverageSquareMetrePriceAgent/avgsqmpriceagent/kg_operations/tsclient.py

```diff
@@ -0,0 +1,86 @@
+################################################
+# Authors: Markus Hofmeister (mh807@cam.ac.uk) #    
+# Date: 23 Oct 2022                            #
+################################################
+
+# The purpose of this module is to provide functionality to use
+# the TimeSeriesClient from the JPS_BASE_LIB
+
+from contextlib import contextmanager
+
+import agentlogging
+from avgsqmpriceagent.errorhandling.exceptions import TSException
+from avgsqmpriceagent.kg_operations.javagateway import jpsBaseLibGW
+from avgsqmpriceagent.datamodel.data import TIMECLASS
+from avgsqmpriceagent.utils.stack_configs import DB_URL, DB_USER, DB_PASSWORD
+
+# Initialise logger instance (ensure consistent logger level with `entrypoint.py`)
+logger = agentlogging.get_logger('prod')
+
+
+class TSClient:
+
+    # Create ONE JVM module view on class level and import all required java classes
+    jpsBaseLibView = jpsBaseLibGW.createModuleView()
+    jpsBaseLibGW.importPackages(jpsBaseLibView, "uk.ac.cam.cares.jps.base.query.*")
+    jpsBaseLibGW.importPackages(jpsBaseLibView, "uk.ac.cam.cares.jps.base.timeseries.*")
+
+    def __init__(self, kg_client, timeclass=TIMECLASS, rdb_url=DB_URL, 
+                 rdb_user=DB_USER, rdb_password=DB_PASSWORD):
+        """
+        Initialise TimeSeriesClient (default properties taken from environment variables)
+
+        Arguments:
+            kg_client (KGClient): KGClient object (as per `kgclient.py` (i.e. PySparqlClient))
+            timeclass: Java time class objects supported by PostgreSQL
+                       (see: https://www.jooq.org/javadoc/dev/org.jooq/org/jooq/impl/SQLDataType.html)
+            rdb_url (str): URL of relational database
+            rdb_user (str): Username for relational database
+            rdb_password (str): Password for relational database
+        """
+
+        # 1) Create an instance of a RemoteStoreClient (to retrieve RDB connection)
+        try:
+            self.conn = TSClient.jpsBaseLibView.RemoteRDBStoreClient(rdb_url, rdb_user, rdb_password)
```

Comment:
This variable should be renamed `self.conn`.

Reply:
Addressed by [d2c4b29](https://github.com/cambridge-cares/TheWorldAvatar/pull/398/commits/d2c4b29a79ee004e72df5814f734354f4782d85e)